src/UI/Implementation/Component/Panel/Report.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Panel;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+
+/**
+ * Class Panel
+ * @package ILIAS\UI\Implementation\Component\Panel
+ */
+class Report extends Panel implements C\Panel\Report {
+	use ComponentHelper;
+
+	/**
+	 * @var \ILIAS\UI\Component\Panel\Sub[]
+	 */
+	private  $sub_panels;
+
+	/**
+	 * @param string $title
+	 * @param \ILIAS\UI\Component\Panel\Sub[]
+	 */
+	public function __construct($title,$sub_panels) {
```

Comment:
Currently we are kind of sitting between fragmented proposals. It would be helpful for me if you look first at the complete picture before proposing single fragments. Otherwise it is hard for me to work with your suggestions. I end up writing improvements for interface/templates/implementation/tests/examples 3 times.

My current solution is not good (on a structural level even buggy), however it is somewhat a result of fragmented proposals. WithContent was moved around quite a bit during the discussion.

From your current solution I gather that you propose the following:
We have an one interface panel. From my understanding of your proposal, the withTitle and withContent part. All panels would inherit from a base class panel implementing this interface, so no extra interface would be needed, right?

--> I can live with that, note however that if done as proposed, we can not type hint anymore on Panel\Sub in the withContent function of panel.